### PR TITLE
Refactor some code in Lowerer to add a bailout to a helper call instruction

### PR DESCRIPTION
**Pull request by rajatd**

### CODE REVIEW

File: lib/Backend/Lower.cpp [hunk 1]
```diff
@@ -25557,6 +25545,18 @@ Lowerer::InsertLoopTopLabel(IR::Instr * insertBeforeInstr)
...
 }
 
+IR::Instr *
+Lowerer::AddBailoutToHelperCallInstr(IR::Instr * helperCallInstr, BailOutInfo * bailoutInfo, IR::BailOutKind  bailoutKind, IR::Instr * primaryBailoutInstr)
+{
+    helperCallInstr = helperCallInstr->ConvertToBailOutInstr(bailoutInfo, bailoutKind);
+    if (bailoutInfo->bailOutInstr == primaryBailoutInstr)
+    {
+        IR::Instr * instrShare = primaryBailoutInstr->ShareBailOut();
+        LowerBailTarget(instrShare);
```
Comment (agarwal-sandeep) [inline lib/Backend/Lower.cpp]: >instrShare [](start = 24, length = 10)

nit: No need for instrShare as we are not doing anything additional on it. Can directly pass primaryBailoutInstr->ShareBailOut() to LowerBailTarget

Comment (rajatd) [reply lib/Backend/Lower.cpp]: instrShare just makes the shared bailout instruction stand out.

**Comment by rajatd** [comment]
@dotnet-bot test this please

**Comment by rajatd** [comment]
@dotnet-bot test this please

**Comment by rajatd** [comment]
@dotnet-bot test Windows x86_release please

**Pull request merged**